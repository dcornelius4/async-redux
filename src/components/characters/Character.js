import React from 'react';
import PropTypes from 'prop-types';

function Character({ character }) {
  return (
    <section>
      <img src={character.img} />
      <p>{character.name}</p>
    </section>
  );
}

Character.propTypes = {
  character: PropTypes.object.isRequired
};

export default Character;
