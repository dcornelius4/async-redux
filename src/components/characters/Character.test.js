import React from 'react';
import { shallow } from 'enzyme';
import Character from './Character';

describe('Character component', () => {
  it('renders a character', () => {
    const character = {
      name: 'Aang',
      image: '/image'
    };
    const wrapper = shallow(<Character character={character} />);
    expect(wrapper).toMatchSnapshot();
  });
});
