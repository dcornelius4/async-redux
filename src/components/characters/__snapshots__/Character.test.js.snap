// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component renders a character 1`] = `
<section>
  <Link
    to="/undefined"
  >
    <img />
    <p>
      Aang
    </p>
  </Link>
</section>
`;
